src/app/ReadClient.cpp

```diff
@@ -850,8 +937,52 @@ CHIP_ERROR ReadClient::SendSubscribeRequest(ReadPrepareParams & aReadPreparePara
         ReturnErrorOnFailure(err = eventFilters.GetError());
     }
 
-    request.IsFabricFiltered(aReadPrepareParams.mIsFabricFiltered).EndOfSubscribeRequestMessage();
-    ReturnErrorOnFailure(err = request.GetError());
+    ReturnErrorOnFailure(err = request.IsFabricFiltered(aReadPrepareParams.mIsFabricFiltered).GetError());
+
+    bool hasEncodeDataVersionList = false;
```

Comment:
Again, might be better to put this in a helper instead of copy/pasting between read and subscribe.  The part up until we call `request.CreateDataVersionFilters()` can't be common, but everything after that point can....

Reply:
agree, done, thanks